gov/proposals.py

```diff
@@ -0,0 +1,106 @@
+import textwrap
+import asyncio
+import subprocess
+from constants import GOVERNANCE_BUDGET_CHANNEL_ID, GOVERNANCE_CHANNEL_ID
+from config.config import current_governance_id, current_budget_id, update_id_values
+
+proposals = []
+
+ongoing_votes = {}
+
+def get_next_governance_id():
+    global current_governance_id
+    current_governance_id += 1
+    return current_governance_id
+
+def get_next_budget_id():
+    global current_budget_id
+    current_budget_id += 1
+    return current_budget_id
+
+def get_governance_id():
+    return current_governance_id
+
+def get_budget_id():
+    return current_budget_id
+
+async def publish_draft(draft, client):
+    if draft["type"].lower() == "budget":
+        channel_id = int(GOVERNANCE_BUDGET_CHANNEL_ID)
+        current_budget_id = get_next_budget_id()
+        update_id_values(current_budget_id, 'budget')  # Update the budget ID in the config file
+        title = f"**Bloom Budget Proposal (BBP) #{current_budget_id}: {draft['name']}**"
+    else:
+        channel_id = int(GOVERNANCE_CHANNEL_ID)
+        current_governance_id = get_next_governance_id()
+        update_id_values(current_governance_id, 'governance')  # Update the governance ID in the config file
+        title = f"**Bloom Improvement Proposal (BIP) #{current_governance_id}: {draft['name']}**"
+
+    channel = client.get_channel(channel_id)
+
+    if not channel:
+        print("Error: Channel not found.")
+        return
+
+    # Create the message content using the draft information
+    msg = f"""
+    {title}
+
+    __**Abstract**__
+    {draft["abstract"]}
+
+    **__Background__**
+    {draft["background"]}
+
+    ** <:inevitable_bloom:1178256658741346344> Yes**
+    ** <:bulby_sore:1127463114481356882> Reassess**
+    ** <:pepe_angel:1161835636857241733> Abstain**
+
+    Vote will conclude in 48h from now.
+    """
+
+    vote_message = await channel.send(textwrap.dedent(msg))
+
+    # Store the vote message ID and relevant information
+    ongoing_votes[vote_message.id] = {
+        "draft": draft,
+        "end_time": asyncio.get_event_loop().time() + 48 * 3600,  # 48 hours in seconds
+        "yes_count": 0,
+        "reassess_count": 0,
+        "abstain_count": 0,
+    }
+
+    # Start the timer coroutine
+    asyncio.create_task(vote_timer(vote_message.id, client, channel_id, title))
+    
+async def vote_timer(message_id, client, channel_id, title):
+    # Sleep until the vote ends
+    await asyncio.sleep(ongoing_votes[message_id]["end_time"] - asyncio.get_event_loop().time())
```

Comment:
Has this been tested?
looks sus 
Why are you not just sleeping for x amount of time assuming this is what you want to do.

Reply:
> Has this been tested? looks sus Why are you not just sleeping for x amount of time assuming this is what you want to do.

I created a fork and merged all three branches and this side of things is working. 

The two issues I found which are yet to be addressed from this testing are:

same variable FILE_PATH used twice across the branches,
message processing needs to be combined instead of separate across emotes and gov.

Reply:
You're right this should just sleep for x amount of time. 

I was using the time comparison for the time to be precise, but unless there is a significant delay for some reason this would be negligible. I will update this.